### DISCUSSION TITLE
时间转日期创建物化视图

### DISCUSSION THREAD

**Discussion by coderjiang** [apache/doris#7082] [A - General / Q&A]
ERROR 1064 (HY000): errCode = 2, detailMessage = The materialized view only support the single column or function expr. Error column: CAST(`purchase_date_local` AS DATE)

**Comment by coderjiang** [comment]
SQL:
create materialized view day_quantity_amount as select CAST(datetime AS DATE) as date,sum(quantity) as quantity,sum(price) as amount from report1 where datetime>='2020-01-01' and datetime<'2022-01-01' group by date;

**Reply by coderjiang** [reply]
why ?

**Reply by Tongyaoyu** [reply]
同问，物化视图中的时间类型需要变成日期类型该怎么办，同理AVG、以及除法运算都不支持，该如何处理？

**Reply by englefly** [reply]
物化试图因为更新成本问题，所以只能处理单行增量更新的计算。 比如求一个sum，新增一行数据，我们就可以把原来的sum和新数据求和。但求AVG，我们没法从原来的AVG和新行求出新的AVG。
